lisa/wa_results_collector.py

```diff
@@ -492,6 +492,48 @@ def row_filter(r): return True
 
         return ret
 
+    def _read_energy_instrument_metrics(self, path):
+        """
+        Parse the WA energy_measurement output.
+        """
+        path_cache = path + ".parquet"
+
+        if not os.path.isfile(path_cache):
+            try:
+                df = pd.read_csv(path)
+            except pandas.errors.ParserError as e:
+                return None
+            df.to_parquet(path_cache)
+        else:
+            df = pd.read_parquet(path_cache)
+
+        if 'device_power' in df.columns:
+            # Looks like this is from an ACME
+
+            df = pd.DataFrame({'value': df['device_power']})
+
+            # Figure out what to call the sample metrics. If the
+            # artifact name has something extra, that will be the
+            # channel (IIO device) name. Use that to differentiate where
+            # the samples came from. If not just call it
+            # 'device_power_sample'.
+            device_name = artifact_name[len('energy_instrument_output') + 1:]
+            name_extra = device_name or 'device'
+            df.loc[:, 'metric'] = '{}_power_sample'.format(name_extra)
+            df.loc[:, 'units'] = 'watts'
+
+        elif 'output_power' in df.columns and 'USB_power' in df.columns:
+            # Looks like this is from a Monsoon
+            # For monsoon the USB and device power are collected
+            # together with the same timestamps, so we can just add them
+            # up.
+            power_samples = df['output_power'] + df['USB_power']
+            df = pd.DataFrame({'value': power_samples})
+            df.loc[:, 'metric'] = 'device_power_sample'
```

Comment:
```
 df.loc[:, 'metric'] = ...
```
Can be turned into:
```
 df['metric'] = ...
```

Reply:
As this code has been moved, I changed it into the follow-up commit which fixes the `.loc` for all the found occurrences within this class.